flow/connectors/mysql/validate.go

```diff
@@ -1,14 +1,172 @@
 package connmysql
 
+
 import (
 	"context"
+	"database/sql"
 	"errors"
+	"fmt"
+	"strings"
 
 	"github.com/go-mysql-org/go-mysql/mysql"
+	_ "github.com/go-sql-driver/mysql"
 
+	"github.com/PeerDB-io/peerdb/flow/connectors/utils"
 	"github.com/PeerDB-io/peerdb/flow/generated/protos"
+	"github.com/PeerDB-io/peerdb/flow/shared"
 )
 
+type MySQLConnector struct {
+	conn *sql.DB
+	config *MySQLConfig
+}
+
+func (c *MySQLConnector) CheckSourceTables(ctx context.Context, tableNames []*utils.SchemaTable) error {
+	if c.conn == nil {
+		return errors.New("check tables: conn is nil")
```

Comment:
MySql connector doesn't maintain a single long lived connection, let Execute manage this